aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SynchronousStorageAdapter.java

```diff
@@ -0,0 +1,279 @@
+     * @return A SynchronousStorageAdapter configured to proxy towards the provided async storage adapter
+     */
+    public static SynchronousStorageAdapter instance(@NonNull LocalStorageAdapter localStorageAdapter) {
+        Objects.requireNonNull(localStorageAdapter);
+        return new SynchronousStorageAdapter(localStorageAdapter, DEFAULT_OPERATION_TIMEOUT_MS);
+    }
+
+    /**
+     * Creates a new instance which will proxy calls to the provided {@link LocalStorageAdapter}.
+     * The synchronous operations exposed by the returned adapter will timeout after the provided
+     * amount of time, in milliseconds.
+     * @param localStorageAdapter Adapter to which calls will be proxied
+     * @param operationTimeoutMs Amount of time after which an operation will time out
+     * @return A SynchronousStorageAdapter configured to proxy towards the provided async storage adapter
+     */
+    public static SynchronousStorageAdapter instance(
+            @NonNull LocalStorageAdapter localStorageAdapter, long operationTimeoutMs) {
+        Objects.requireNonNull(localStorageAdapter);
+        return new SynchronousStorageAdapter(localStorageAdapter, operationTimeoutMs);
+    }
+
+    /**
+     * Initializes the storage adapter.
+     * @param context An Android Context
+     * @return The list of model schema that are available for use in the adapter
+     * @throws DataStoreException On any initialization failure
+     */
+    @SuppressWarnings("UnusedReturnValue")
+    List<ModelSchema> initialize(@NonNull Context context) throws DataStoreException {
+        return Await.result(
+            operationTimeoutMs,
+            (Consumer<List<ModelSchema>> onResult, Consumer<DataStoreException> onError) ->
+                localStorageAdapter.initialize(context, onResult, onError)
+        );
+    }
+
+    /**
+     * Terminate use of the storage adapter.
+     * @throws DataStoreException On failure to terminate
+     */
+    void terminate() throws DataStoreException {
+        localStorageAdapter.terminate();
+    }
+
+    /**
+     * Save a model into the storage adapter.
+     * @param model Model to save
+     * @param <T> Type of model being saved
+     * @throws DataStoreException On any failure to save model into storage adapter
+     */
+    <T extends Model> void save(@NonNull T model) throws DataStoreException {
+        //noinspection ConstantConditions
+        save(model, null);
+    }
+
+    /**
+     * Save a model.
+     * @param model Model to save
+     * @param predicate An existing instance of the model in the storage adapter must meet these criteria
+     *                  in order for the save to succeed. If null, no criteria are considered
+     * @param <T> Type of model being saved
+     * @throws DataStoreException On any failure to save the model
+     */
+    <T extends Model> void save(@NonNull T model, @NonNull QueryPredicate predicate)
```

Comment:
Let's make predicate `@Nullable` since the overloaded method calls this with `null` predicate.

Reply:
(See discussion above.)